tests/acceptance/10_files/unsafe/can_set_sticky_bits_with_nonroot_owner_and_group.cf

```diff
@@ -0,0 +1,105 @@
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+
+#######################################################
+
+bundle agent init
+{
+  vars:
+    "users" slist => { "user01", "user02" };
+    "modes" slist => {
+                       "2700",
+                       "2750",
+                       "2755",
+                       "2770",
+                       "2771",
+                       "2775",
+                       "2751",
+                      };
+
+  users:
+    "$(users)"
+      policy => "present";
+
+}
+
+bundle agent cleanup
+{
+  users:
+    "$(init.users)"
+      policy => "absent",
+      comment => "Purge the users we created to test with";
+}
+
+#######################################################
+
+bundle agent test
+{ 
+  meta:
+    "description" string => "Test that sgid can be set when ownership is non root";
+
+    "test_skip_unsupported"
+      string => "windows",
+      meta => { "redmine7989", "zendesk2797" };
+
+  files:
+    linux::
+      "$(G.testdir)/test_sgid/$(init.modes)/."
+      create => "true",
+      perms => test_perms( $(init.modes), "user01", "user02"),
+      classes => scoped_classes_generic("namespace", "file_$(init.modes)"),
+      comment => "Make sure a directory named for each desired mode we want to test exists.";
+}
+
+body perms test_perms(m, o, g)
+{
+      mode => "$(m)";
+      owners => { "$(o)" };
+      groups => { "$(g)" };
+}
+
+#######################################################
+
+bundle agent check
+{
+
+  vars:
+      # We expect to find an ok_ prefixed class for each of the modes defined in
+      # init.
+      "expected_classes" slist => maplist("ok_$(this)", @(init.modes));
+
+      # Get the octal mode for each of the files
+      "result_permoct_$(init.modes)"
+        string => execresult('/usr/bin/stat -c %a $(G.testdir)/test_sgid/$(init.modes)', "useshell"),
```

Comment:
AIX (and possibly others) doesn't have this tool. Can you add it to the list of tools near the top of `dcs.cf.sub`, and then use `"test_skip_unsupported" string => "windows|!has_stat";` to skip it there?